### PR TITLE
feat(Bonus Pagamenti Digitali): [#176578587] Cobadge detail screen

### PR DESCRIPTION
**Pull request by debiff**
## Short description
This PR add the detail of a Cobadge credit card

## List of changes proposed in this pull request
- Add the Cobadge detail screen
- Add Cobadge card component

### CODE REVIEW

File: ts/features/wallet/cobadge/component/CobadgeCard.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+const styles = StyleSheet.create({
+  abiLogoFallback: {
+    width: 40,
+    height: 40,
+    resizeMode: "contain"
+  },
+  brandLogo: {
+    width: 80,
+    height: 40,
+    resizeMode: "contain"
```
Comment (fabriziofff) [inline ts/features/wallet/cobadge/component/CobadgeCard.tsx:34]:
```suggestion
    width: 48,
    height: 30,
    resizeMode: "contain"
```

Changed following the Invision specs:
Schermata 2021-01-29 alle 18 18 35

File: ts/features/wallet/cobadge/component/CobadgeCard.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+              source={{ uri: props.abiLogo }}
+              style={imageStyle}
+              testID={"abiLogo"}
+            />
+          ) : (
+            <Image
+              source={abiLogoFallback}
+              style={styles.abiLogoFallback}
+              testID={"abiLogoFallback"}
+            />
```
Comment (fabriziofff) [inline ts/features/wallet/cobadge/component/CobadgeCard.tsx:68]: On iOS I always see the fallback image. 


https://user-images.githubusercontent.com/26501317/106308043-70930d80-6260-11eb-8e70-bb2f54612a7c.mov

```suggestion
          {props.abiLogo && imageStyle ? (
            <Image
              source={{ uri: props.abiLogo }}
              style={imageStyle}
              testID={"abiLogo"}
              key={"abiLogo"}
            />
          ) : (
            <Image
              source={abiLogoFallback}
              style={styles.abiLogoFallback}
              testID={"abiLogoFallback"}
              key={"abiLogoFallback"}
            />
```
My guess is that the node occupies the same position and the cache is not refreshered, obtaining something similar to https://github.com/facebook/react-native/issues/9195. 
Using different keys it renders correctly for me both on iOS and Android.

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx [hunk 3]
```diff
@@ -62,18 +67,22 @@ const CobadgeWalletPreview: React.FunctionComponent<Props> = props => {
...
   );
+
+  const brandLogo = getCardIconFromBrandLogo(props.cobadge.info);
   return (
     <CardPreview
       left={renderLeft(props, imgDimensions)}
-      image={getCardIconFromBrandLogo(props.cobadge.info)}
-      onPress={() => props.navigateToCobadgeDetails(props.cobadge)}
+      image={brandLogo}
+      onPress={() => props.navigateToCobadgeDetails(props.cobadge, brandLogo)}
```
Comment (fabriziofff) [inline ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx:76]: What you think about passing only `props.cobadge` as navigation param? The brandlogo can always be uniquely derived using `getCardIconFromBrandLogo`.

File: ts/features/wallet/cobadge/component/__test__/CobadgeCard.test.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import defaultCardIcon from "../../../../../../img/wallet/cards-icons/unknown.png";
+
+jest.mock("../../../onboarding/bancomat/screens/hooks/useImageResize");
+
+const aCaption = "****1234";
+const anAbiLogo = "http://127.0.0.1:3000/static_contents/logos/abi/03069.png";
+const aBrandLogo = defaultCardIcon;
+const anExpireMonth = "6";
+const anExpireYear = "2021";
+describe("BPayWalletPreview component", () => {
```
Comment (fabriziofff) [inline ts/features/wallet/cobadge/component/__test__/CobadgeCard.test.tsx:20]:
```suggestion
describe("CoBadgeWalletPreview component", () => {
```

**Review by fabriziofff — CHANGES_REQUESTED** [review]

File: ts/features/wallet/cobadge/component/__test__/CobadgeCard.test.tsx (hunk 4 above)
Comment (debiff) [reply ts/features/wallet/cobadge/component/__test__/CobadgeCard.test.tsx]: Fixed in [0280969](https://github.com/pagopa/io-app/pull/2751/commits/028096919fd76eea38b2c405553e497075326f52)

File: ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx (hunk 3 above)
Comment (debiff) [reply ts/features/wallet/cobadge/component/CobadgeWalletPreview.tsx]: Yes, sure.
Modified in [0816b1d](https://github.com/pagopa/io-app/pull/2751/commits/0816b1d1cca33008d4900f323e424274e7f2fb49)

File: ts/features/wallet/cobadge/component/CobadgeCard.tsx (hunk 2 above)
Comment (debiff) [reply ts/features/wallet/cobadge/component/CobadgeCard.tsx]: Thanks for the suggestion. Working always on the same card I never notice this bug.
Fixed in [a6c4e84](https://github.com/pagopa/io-app/pull/2751/commits/a6c4e84ab2755f11f24c1a0262a820fc877be78c)

File: ts/features/wallet/cobadge/component/CobadgeCard.tsx (hunk 1 above)
Comment (debiff) [reply ts/features/wallet/cobadge/component/CobadgeCard.tsx]: Modified dimensions accordingly to the InVision specs in [8e95104](https://github.com/pagopa/io-app/pull/2751/commits/8e9510497ff5b0c333b56f55bfbeb568b8a6c38c)

**Review by fabriziofff — APPROVED** [review]

**Pull request merged**